moment_kinetics/test/calculus_tests.jl

```diff
@@ -1565,6 +1566,297 @@ function runtests()
+                     (2, 14, 4.e-7),
+                     (2, 15, 5.e-8),
+                     (2, 16, 2.e-8),
+                     (2, 17, 5.e-9),
+                     
+                     (3, 6, 2.e-3),
+                     (3, 7, 2.e-4),
+                     (3, 8, 5.e-5),
+                     (3, 9, 3.e-6),
+                     (3, 10, 2.e-6),
+                     (3, 11, 3.e-7),
+                     (3, 12, 6.e-8),
+                     (3, 13, 9.e-9),
+                     (3, 14, 2.e-9),
+                     (3, 15, 4.e-10),
+                     (3, 16, 3.e-11),
+                     (3, 17, 1.e-11),
+                     
+                     (4, 5, 1.e-3),
+                     (4, 6, 8.e-5),
+                     (4, 7, 3.e-5),
+                     (4, 8, 3.e-6),
+                     (4, 9, 6.e-7),
+                     (4, 10, 8.e-8),
+                     (4, 11, 2.e-8),
+                     (4, 12, 3.e-9),
+                     (4, 13, 3.e-10),
+                     (4, 14, 5.e-11),
+                     (4, 15, 3.e-12),
+                     (4, 16, 2.e-12),
+                     (4, 17, 1.e-12),
+                     
+                     (5, 5, 4.e-4),
+                     (5, 6, 3.e-5),
+                     (5, 7, 6.e-6),
+                     (5, 8, 4.e-7),
+                     (5, 9, 9.e-8),
+                     (5, 10, 4.e-9),
+                     (5, 11, 2.e-9),
+                     (5, 12, 8.e-11),
+                     (5, 13, 3.e-11),
+                     (5, 14, 1.e-12),
+                     (5, 15, 4.e-13),
+                     (5, 16, 4.e-13),
+                     (5, 17, 2.e-12),
+                     )
+
+                # define inputs needed for the test
+                L = 6.0
+                bc = "zero"
+                element_spacing_option = "uniform"
+                # create the coordinate struct 'x'
+                # This test runs effectively in serial, so implicitly uses
+                # `ignore_MPI=true` to avoid errors due to communicators not being fully
+                # set up.
+                x, spectral = define_test_coordinate("vperp"; ngrid=ngrid,
+                                                     nelement=nelement, L=L,
+                                                     discretization="gausslegendre_pseudospectral",
+                                                     bc=bc,
+                                                     element_spacing_option=element_spacing_option,
+                                                     collision_operator_dim=false)
+                expected_f = @. exp(-x.grid^2)
+                d2f = @. 4.0*(x.grid^2 - 1.0)*exp(-x.grid^2)
+                # create array for the numerical solution
+                f = similar(expected_f)
+                # create array for RHS vector b
+                b = similar(expected_f)
+                # solve for f
+                mul!(b,spectral.mass_matrix,d2f)
```

Comment:
`d2f` in other places would mean `d^2f/dx^2`, so I think it's clearer to use a different variable name. Also could use a comment on what equation is solved
```suggestion
                # Test solver for
                #   Laplacian_f = 1/x * d/dx(x*df/dx)
                Laplacian_f = @. 4.0*(x.grid^2 - 1.0)*exp(-x.grid^2)
                # create array for the numerical solution
                f = similar(expected_f)
                # create array for RHS vector b
                b = similar(expected_f)
                # solve for f
                mul!(b,spectral.mass_matrix,Laplacian_f)
```